metal/devices.go

```diff
@@ -111,7 +132,12 @@ func nodeAddresses(device *packngo.Device) ([]v1.NodeAddress, error) {
 				privateIP = address.Address
 				addrType = v1.NodeInternalIP
 			}
-			addresses = append(addresses, v1.NodeAddress{Type: addrType, Address: address.Address})
+			addr = v1.NodeAddress{Type: addrType, Address: address.Address}
+
+			if _, ok := unique[addr]; !ok {
```

Comment:
We could take advantage of this CloudProvider helper and avoid some of the logic here

https://github.com/kubernetes/cloud-provider/blob/9a26d4868f3a005a0cb22e19c189499642a29b04/node/helpers/address.go#L27-L42

Reply:
Instead of the whole `unique` thing? I guess so. I don't really care that much. It replaces 4 lines of code with 1 line of code 3 times, which really isn't all that much. It also is inefficient, looping each time (which doesn't matter all that much in the scale of a few entries). But it also eliminates our ability to do the `unique` change I suggested above to fit your requirements.